bigquery_etl/view/__init__.py

```diff
@@ -0,0 +1,250 @@
+"""Represents a SQL view."""
+
+import attr
+import sqlparse
+import string
+import time
+
+from google.api_core.exceptions import BadRequest
+from google.cloud import bigquery
+from pathlib import Path
+
+from bigquery_etl.format_sql.formatter import reformat
+from bigquery_etl.metadata.parse_metadata import DatasetMetadata, DATASET_METADATA_FILE
+from bigquery_etl.util import extract_from_query_path
+from bigquery_etl.schema import Schema
+
+
+# skip validation for these views
+SKIP_VALIDATION = {
+    # not matching directory structure, but created before validation was enforced
+    "sql/moz-fx-data-shared-prod/stripe/subscription/view.sql",
+    "sql/moz-fx-data-shared-prod/stripe/product/view.sql",
+    "sql/moz-fx-data-shared-prod/stripe/plan/view.sql",
+    "sql/moz-fx-data-shared-prod/telemetry/client_probe_counts_v1/view.sql",
+    "sql/moz-fx-data-shared-prod/telemetry/clients_daily_histogram_aggregates_v1/view.sql",
+    "sql/moz-fx-data-shared-prod/telemetry/clients_scalar_aggregates_v1/view.sql",
+    "sql/moz-fx-data-shared-prod/telemetry/clients_daily_scalar_aggregates_v1/view.sql",
+    "sql/moz-fx-data-shared-prod/telemetry/clients_histogram_aggregates_v1/view.sql",
+    "sql/moz-fx-data-shared-prod/telemetry/clients_probe_processes/view.sql",
+    # tests
+    "sql/moz-fx-data-test-project/test/simple_view/view.sql",
+}
+
+# skip publishing these views
+SKIP_PUBLISHING = {
+    # Access Denied
+    "activity_stream/tile_id_types/view.sql",
+    "pocket/pocket_reach_mau/view.sql",
+    "telemetry/buildhub2/view.sql",
+    # Dataset glam-fenix-dev:glam_etl was not found
+    # TODO: this should be removed if views are to be automatically deployed
+    *[str(path) for path in Path("sql/glam-fenix-dev").glob("glam_etl/**/view.sql")],
+    # tests
+    "sql/moz-fx-data-test-project/test/simple_view/view.sql",
+}
+
+# suffixes of datasets with non-user-facing views
+NON_USER_FACING_DATASET_SUFFIXES = (
+    "_derived",
+    "_external",
+    "_bi",
+    "_restricted",
+)
+
+
+@attr.s(auto_attribs=True)
+class View:
```

Comment:
This class combines some of the code related to managing views and makes it easier to reuse some of the logic (e.g. for validation). I'm thinking of doing something similar for queries since currently the logic is spread across several scripts with lots of redundant code that is hard to test.